### PR TITLE
Move files offline

### PR DESCRIPTION
**Pull request by TristanWright**
for #378 

- [x] create file selection
- [x] `/file/move` bulk move endpoint
- [x] disable ui when moving files in progress

### CODE REVIEW

**Comment by cjh1** [comment]
@TristanWright The result on my discussion on the Girder list is that we should add a custom endpoint to our plugin that would do a self.model('upload').moveFileToAssetstore(...) on a set of files that get passed to it. We would probably also want to add the option to pass in a ProgressContext object and update it with the count of bytes transferred, so that we can relay progress to the UI.

**Comment by TristanWright** [comment]
latest test uses 'moveFilesToAssetStore' my girder version is still 1.6, do I need to scrap vm to get it up to 2.1?

**Comment by cjh1** [comment]
Should be able to do the following:

```
cd /opt/hpccloud/girder/
git pull origin master
girder-install web
```

**Comment by TristanWright** [comment]
Because the ProgressContext chunk size is so big there are about as many updates as there are files. Is there a way to adjust it? You can preview it right now if you'd like to see what I mean.

**Comment by TristanWright** [comment]
Also, my file move test is working on 2.7 but not 3.4?

**Comment by cjh1** [comment]
Looks like you have some print statements that use the 2.7 syntax

```
 print tempfile.gettempdir()
```

They can probably be remove anyway?

**Comment by TristanWright** [comment]
I'm weak on python 2->3 differences, is there anything else that could be causing a failure?

**Comment by cjh1** [comment]
We are using six to maintain compatibility between python 2.7 and 3. This is a good resource:

https://pythonhosted.org/six/

In this case you want to add a list( ) round your map. In 2.7 map returned a list, not in 3.

**Comment by TristanWright** [comment]
@cjh1 this is ready.

**Comment by TristanWright** [comment]
So the file is definitely moved, I can see it on the file system, the local assetstore is of filesystem type. 

The girder exception is being thrown from here: 
https://github.com/girder/girder/blob/57231f3984e268822359d72320580dac439093e2/girder/utility/filesystem_assetstore_adapter.py#L261

**Comment by cjh1** [comment]
@TristanWright So does the path not match what you are seeing on the filesystem?

**Comment by TristanWright** [comment]
The assetstore path does match and so does the file's path.

**Comment by cjh1** [comment]
So if ```os.path.isfile``` is false what is the type of the path?

**Comment by TristanWright** [comment]
So with some logging I've found that the assetstore path is not being appended in when path is passed to `fullPath()`

**Comment by TristanWright** [comment]
because `file.get('imported')` is `True`

**Comment by TristanWright** [comment]
After the girder changes this is ready to merge, however I'm getting a ui lock when the file is loading. 

screen shot 2017-01-19 at 11 43 32

Looking into this...

**Comment by cjh1** [comment]
@TristanWright Things lock up when you try to load the moved file?

**Comment by TristanWright** [comment]
Any file, this is happening before the cancel request changes too, but I don't remember this ever happening before, that is I was able to test the cancel request changes. I'm running Chrome 55.0.2883.95

**Comment by TristanWright** [comment]
I'm wondering if it has to do with inserting 200KB of text to the DOM?

**Comment by cjh1** [comment]
So this happens on master as well?

**Comment by TristanWright** [comment]
Yes I'm seeing it there too, are you? Basically when the file is loading I cannot close the modal.

**Comment by cjh1** [comment]
Let try

**Comment by cjh1** [comment]
Works for me on master with a 249.8 KB file.

**Comment by TristanWright** [comment]
I've restarted chrome and it's fine now, very interesting. This can be merged.

**Comment by cjh1** [comment]
Before we merge can we add to the test case check that we can download the file after its moved? This would catch the 'imported' issue we where seeing earlier in the week.

**Comment by TristanWright** [comment]
Def. I'll get append that to the current test

File: server/hpccloud/server/file.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+def move_files_to_assetstore(files, totalsize):
+    user, token = getCurrentUser(True)
+    assetstore = ModelImporter.model('assetstore').getCurrent()
+    upload = ModelImporter.model('upload')
+    with ProgressContext(True, interval=1.0, message=files[0]['name'],
+                         user=user, token=token,
+                         total=totalsize, title='Moving files') as ctx:
+        for file in files:
+            ctx.update(message=file['name'])
```
Comment (cjh1) [inline server/hpccloud/server/file.py]: Do these messages get displayed in the UI or are we just using the size to track progress?

**Comment by cjh1** [comment]
@TristanWright Tested this out, works well, great job! Two things:

- Now that have cancellation (#545) I think we can add more cancel calls on umount?
- Will this move functionality work in the other workflows?

File: server/hpccloud/server/file.py (hunk 1 above)
Comment (TristanWright) [inline server/hpccloud/server/file.py]: You're right we're not displaying it anywhere. I'm not sure this line has the intended effect either.

**Comment by TristanWright** [comment]
> Now that have cancellation (#545) I think we can add more cancel calls on umount?

You mean for other components that have some call? Yes, `client.cancel()` will cancel the most recent active request if there is any. When adding this I was wondering if it should be expanded to _all_ requests, but I don't think we have many spots where we're doing simultaneous calls?

> Will this move functionality work in the other workflows?

We should be able to integrate it. @jourdain is doing an overhaul of the workflow ui (#579) which will make it easier too.

**Comment by cjh1** [comment]
OK, lets go ahead and merge, we can raise a follow up issue to ensure it works on other workflows

**Pull request merged**